### DISCUSSION TITLE
guide/introduction/about

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#946] [Announcements]
# guide/introduction/about

关于 提示 我一直想开发一款后台管理系统，虽然看过许多优秀的开源项目，但始终没有找到合适的解决方案。因此，利用空闲时间，我开始着手自主开发，最终诞生了 码动。 码动 前端开发采用 Vue3.0、Vite、 TypeScript 的中后台解决方案，目标是为开发中大型项目提供开箱即用的解决方案。包括二次封装组件、utils、hooks、动态菜单、权限校验、...

https://docs.madong.tech/guide/introduction/about.html

**Comment by xiaodai1992625** [comment]
我php版本8.1执行composer一堆报错 emmmm'

**Comment by imran-siddique** [comment]
Thanks @sunilp — really thoughtful feedback. Let me address each point:

**1. Interface alignment**
- Yes, \ADKPolicyEvaluator\ returns \PolicyDecision\ with structured \
eason\, \matched_rule\ (which policy triggered), \erdict\ (allow/deny/escalate), and \metadata\ dict for arbitrary context. The audit trail captures which rule denied and why.
- Yes, \DelegationScope.narrow()\ enforces monotonic narrowing — child tools must be subset of parent, max_tool_calls can only decrease, max_depth decrements, and \
ead_only\ is sticky (once true, can't go back to false). This is tested in the suite.

**2. Additional hooks** — great suggestions:
- \efore_model_send\ — agreed this is the natural next enforcement point. We have prompt injection detection (\prompt_injection.py\) and PII redaction (\mute_agent.py\) that would slot in here. Will add once ADK exposes this hook.
- \on_tool_error\ / \on_agent_error\ — good call for audit completeness. Will add these to the next iteration.

**3. Packaging** — agreed, standalone PyPI package. Will publish as \dk-agentmesh\ so it's independently installable.

**On PolicyDecision schema alignment** — this is the key insight. If we align the return types, any \PolicyEvaluator\ backend (APS, AGT, simple YAML, LDP) becomes a drop-in replacement. I'll match the field names to what you have in the GovernancePlugin. The current schema:

\\\python
@dataclass
class PolicyDecision:
    verdict: Verdict  # allow | deny | escalate
    reason: str
    matched_rule: str
    timestamp: datetime
    metadata: dict[str, Any]
\\\

Does this align with what you and @aeoess have? Happy to adjust fields.

Will publish to PyPI and share here once it's up.
